plugins/inputs/aliyuncms/aliyuncms.go

```diff
@@ -326,71 +380,73 @@ func (s *AliyunCMS) updateWindow(relativeTo time.Time) {
-		resp, err := s.client.DescribeMetricList(req)
-		if err != nil {
-			return errors.Errorf("failed to query metricName list: %v", err)
-		} else if resp.Code != "200" {
-			s.Log.Errorf("failed to query metricName list: %v", resp.Message)
-			break
-		}
-
-		var datapoints []map[string]interface{}
-		if err = json.Unmarshal([]byte(resp.Datapoints), &datapoints); err != nil {
-			return errors.Errorf("failed to decode response datapoints: %v", err)
-		}
+	for _, region := range s.Regions {
+		req := cms.CreateDescribeMetricListRequest()
+		req.Period = strconv.FormatInt(int64(time.Duration(s.Period).Seconds()), 10)
+		req.MetricName = metricName
+		req.Length = "10000"
+		req.Namespace = s.Project
+		req.EndTime = strconv.FormatInt(s.windowEnd.Unix()*1000, 10)
+		req.StartTime = strconv.FormatInt(s.windowStart.Unix()*1000, 10)
+		req.Dimensions = metric.requestDimensionsStr
+		req.RegionId = region
+
+		for more := true; more; {
+			resp, err := s.client.DescribeMetricList(req)
+			if err != nil {
+				return errors.Errorf("failed to query metricName list: %v", err)
+			} else if resp.Code != "200" {
+				s.Log.Errorf("failed to query metricName list: %v", resp.Message)
+				break
+			}
 
-		if len(datapoints) == 0 {
-			s.Log.Debugf("No metrics returned from CMS, response msg: %s", resp.Message)
-			break
-		}
+			var datapoints []map[string]interface{}
+			if err = json.Unmarshal([]byte(resp.Datapoints), &datapoints); err != nil {
+				return errors.Errorf("failed to decode response datapoints: %v", err)
+			}
 
-	NextDataPoint:
-		for _, datapoint := range datapoints {
-			fields := map[string]interface{}{}
-			datapointTime := int64(0)
-			tags := map[string]string{}
-			for key, value := range datapoint {
-				switch key {
-				case "instanceId", "BucketName":
-					tags[key] = value.(string)
-					if metric.discoveryTags != nil { //discovery can be not activated
-						//Skipping data point if discovery data not exist
-						if _, ok := metric.discoveryTags[value.(string)]; !ok &&
-							!metric.AllowDataPointWODiscoveryData {
-							s.Log.Warnf("Instance %q is not found in discovery, skipping monitoring datapoint...", value.(string))
-							continue NextDataPoint
-						}
+			if len(datapoints) == 0 {
+				s.Log.Debugf("No metrics returned from CMS, response msg: %s", resp.Message)
+				break
+			}
 
-						for k, v := range metric.discoveryTags[value.(string)] {
-							tags[k] = v
+		NextDataPoint:
+			for _, datapoint := range datapoints {
+				fields := map[string]interface{}{}
+				datapointTime := int64(0)
+				tags := map[string]string{}
+				for key, value := range datapoint {
+					switch key {
+					case "instanceId", "BucketName":
+						tags[key] = value.(string)
+						if metric.discoveryTags != nil { //discovery can be not activated
+							//Skipping data point if discovery data not exist
+							if _, ok := metric.discoveryTags[value.(string)]; !ok &&
+								!metric.AllowDataPointWODiscoveryData {
```

Comment:
I think
```suggestion
							_, ok := metric.discoveryTags[value.(string)]
							if !ok && !metric.AllowDataPointWODiscoveryData {
```
will increase readability. :-)

Reply:
Agree, fixed